### PR TITLE
Remove strdup and strcasecmp from codebase

### PR DESCRIPTION
**Pull request by rogersm**
Removed strdup and stracasecmp from the codebase. They've been part of stdc for ever and it makes no sense to keep using the ones included in the mud.

### CODE REVIEW

File: doc/Guildmud Persistance model.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+A simple proposal for a persistance model.
+
+Using @zachflower [PR9](https://github.com/mudcoders/guildmud/pull/9) as a guideline, let's implement data persistance in sqlite.
+
+We have three options to persist the data in sqlite, from simpler to most complex:
+
+1. Give an id to every data structure in the system, and store them in a single table. Content of the data structure is the persisted data structure (persistance model to be defined)
+
+![DB Model](file:///Users/rogersm/Dropbox/dev/guildmud/doc/db_model1.png)
```
Comment (zachflower) [inline doc/Guildmud Persistance model.md:9]: These don't seem to be pulling up when previewing the file. Feel free to put the images in the repository and reference them from there.

**Review by zachflower — CHANGES_REQUESTED** [review]
One quick change request after perusal (not related to the code). I'll also be testing on both Vagrant and macOS today (looks like when the Makefile was changed the Travis config wasn't updated, so the successful CI status isn't an accurate representation)

**Comment by zachflower** [comment]
Did a quick test, everything looks good! Just a quick fix to those images and we're good to go 👍

File: doc/Guildmud Persistance model.md (hunk 1 above)
Comment (rogersm) [reply doc/Guildmud Persistance model.md:9]: Why did this get pushed? This is something that was not ready to be published.

Will cancel the PR and send a new one.

**Pull request closed without merging**